app/upgrades/v10/upgrades.go

```diff
@@ -0,0 +1,359 @@
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	icacontrollermigrations "github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/controller/migrations/v6"
+	clientkeeper "github.com/cosmos/ibc-go/v7/modules/core/02-client/keeper"
+	channelkeeper "github.com/cosmos/ibc-go/v7/modules/core/04-channel/keeper"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+	ibctmmigrations "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint/migrations"
+
+	claimkeeper "github.com/Stride-Labs/stride/v9/x/claim/keeper"
+	claimtypes "github.com/Stride-Labs/stride/v9/x/claim/types"
+	icacallbackskeeper "github.com/Stride-Labs/stride/v9/x/icacallbacks/keeper"
+	mintkeeper "github.com/Stride-Labs/stride/v9/x/mint/keeper"
+	minttypes "github.com/Stride-Labs/stride/v9/x/mint/types"
+	ratelimitkeeper "github.com/Stride-Labs/stride/v9/x/ratelimit/keeper"
+	ratelimitgov "github.com/Stride-Labs/stride/v9/x/ratelimit/keeper/gov"
+	ratelimittypes "github.com/Stride-Labs/stride/v9/x/ratelimit/types"
+	recordskeeper "github.com/Stride-Labs/stride/v9/x/records/keeper"
+	recordstypes "github.com/Stride-Labs/stride/v9/x/records/types"
+	stakeibckeeper "github.com/Stride-Labs/stride/v9/x/stakeibc/keeper"
+	stakeibctypes "github.com/Stride-Labs/stride/v9/x/stakeibc/types"
+
+	cosmosproto "github.com/cosmos/gogoproto/proto"
+	deprecatedproto "github.com/golang/protobuf/proto" //nolint:staticcheck
+)
+
+var (
+	UpgradeName     = "v10"
+	EpochProvisions = sdk.NewDec(929_681_506)
+
+	StakingProportion                     = "0.1605"
+	CommunityPoolGrowthProportion         = "0.2158"
+	StrategicReserveProportion            = "0.4879"
+	CommunityPoolSecurityBudgetProportion = "0.1358"
+
+	CommunityPoolGrowthAddress = "stride1lj0m72d70qerts9ksrsphy9nmsd4h0s88ll9gfphmhemh8ewet5qj44jc9"
+	BadKidsCustodian           = "stride17z6yy8vfgklgej9m848jm7rkp270gd9pgaw8zu"
+	BadKidsTransferAmount      = sdk.NewInt(15_000_000_000)
+	Ustrd                      = "ustrd"
+
+	MinInitialDepositRatio = "0.25"
+
+	// airdrop distributor addresses
+	DistributorAddresses = map[string]string{
+		"stride":  "stride1cpvl8yf848karqauyhr5jzw6d9n9lnuuu974ev",
+		"gaia":    "stride1fmh0ysk5nt9y2cj8hddms5ffj2dhys55xkkjwz",
+		"osmosis": "stride1zlu2l3lx5tqvzspvjwsw9u0e907kelhqae3yhk",
+		"juno":    "stride14k9g9zpgaycpey9840nnpa66l4nd6lu7g7t74c",
+		"stars":   "stride12pum4adk5dhp32d90f8g8gfwujm0gwxqnrdlum",
+		"evmos":   "stride10dy5pmc2fq7fnmufjfschkfrxaqnpykl6ezy5j",
+	}
+	NewDistributorAddresses = map[string]string{
+		"stride":  "stride1w02dg74j8s38gqn6mvlr87hkvyv5rgp3cqe9se",
+		"gaia":    "stride1w0w0gr6u796y2mjl9fuqt66jqvk3j59jq3jtpg",
+		"osmosis": "stride1mfg5ck02tlyzdtdpaj70ngtgjs2vuawtkfz7xd",
+		"juno":    "stride1ral4dsqk0nzyqlwtuyxgavfvx8hegml7u0rzx3",
+		"stars":   "stride1rm9nxc5pw3k5r5s6lm85k73mfp734nhnxq570g",
+		"evmos":   "stride1ej4e7x2hanmy6vrzrjh06g6dnfq5kxm73dmgsw",
+	}
+
+	RateLimitDurationHours = uint64(24)
```

Comment:
This seems aggressive - do you know what Osmosis uses?

Reply:
I don't. I agree it's aggressive. Wdyt it should be?